### PR TITLE
Add Language settings Page

### PR DESCRIPTION
**Pull request by hueitan**
to do

- [x] banana reload i18n message
- [x] support all languages
- [x] Confirm which language is shown in language settings (shown first 10)
- [x] change app language settings
- [x] Keep the language setting in localStorage

except the following:

- [ ] change article language settings

### CODE REVIEW

**Comment by hueitan** [comment]
It sesms like banana-i18n setLocale https://github.com/wikimedia/banana-i18n#setting-the-locale doesn't work properly

> All .i18n() calls will set the message for the new locale from there onwards.

I don't see it fetch the new language json during this call.

File: src/index.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { loadMessages } from 'api'
 import { App } from 'components'
 import '../style/style.less'
 
-const lang = navigator.language.substr(0, 2)
+const lang = localStorage.getItem('language-app') || navigator.language.substr(0, 2)
```
Comment (stephanebisson) [inline src/index.js]: Please encapsulate localStorage in a utils.

File: style/language.less [hunk 2]
```diff
@@ -0,0 +1,57 @@
+.language {
+
```
Comment (stephanebisson) [inline style/language.less]: Unnecessary blank line. (we need a less linter)

File: src/components/Language.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    setItems(items.map(item => {
+      item.isSelected = item.lang === lang
+      return item
+    }))
+  }, [lang])
+
+  return [items, query, setQuery]
+}
+
+const languagePopup = ({ close, i18n }) => {
```
Comment (stephanebisson) [inline src/components/Language.js]: components should start with upper case

File: src/components/App.js [hunk 4]
```diff
@@ -1,26 +1,24 @@
 import { h } from 'preact'
 import { useReducer, useState } from 'preact/hooks'
 import { Routes, Softkey, PopupContainer } from 'components'
-import { LanguageContext, I18nContext, SoftkeyContext, PopupContext } from 'contexts'
+import { I18nContext, SoftkeyContext, PopupContext } from 'contexts'
 import { SoftkeyReducer } from 'reducers'
 
-export const App = ({ lang, i18n }) => {
+export const App = ({ i18n }) => {
```
[the remark this answers is not in the document]
Comment (hueitan) [reply src/components/App.js]: remove the prop in `index.js`

File: src/utils/languages.js [hunk 5]
```diff
@@ -0,0 +1,1457 @@
+// json language file given by iOS team
+const languages = [
```
[the remark this answers is not in the document]
Comment (hueitan) [reply src/utils/languages.js]: as per discussion, we will implement the whitelist/blacklist when it is needed.

**Pull request merged**